src/main/java/com/pingcap/tikv/codec/LongUtils.java

```diff
@@ -109,7 +111,7 @@ public static void writeUVarLong(CodecDataOutput cdo, long value) {
      * @return value decoded
      * @exception InvalidCodecFormatException wrong format of binary encoding encountered
      */
-    public static long readLongFully(CodecDataInput cdi) {
+    public static long readLongFully(CodecDataInput cdi) throws IOException {
```

Comment:
Basically we don't have really IOException even though underlining interfaces (DataInput / DataOutput) signatures provide so. You might check ByteStream implementation to prove it. It's preferred to be shadowed in subclass since we don't really need to catch a never-thrown exception. If we need an exception to represent a en/decoding problem, we might want to use something like InvalidCodecFormatException to be more specific.